phase/configure_k0s.go

```diff
@@ -285,13 +286,49 @@ func (p *ConfigureK0s) configFor(h *cluster.Host) (string, error) {
 		cfg = p.newBaseConfig.Dup()
 	}
 
-	var addr string
+	var (
+		sans []string
+		addr string
+	)
+
 	if h.PrivateAddress != "" {
 		addr = h.PrivateAddress
 	} else {
 		addr = h.Address()
 	}
 
+	// convert sans from unmarshaled config into []string
+	oldsans := cfg.Dig("spec", "api", "sans")
+	switch oldsans := oldsans.(type) {
+	case []interface{}:
+		for _, v := range oldsans {
+			if s, ok := v.(string); ok {
+				sans = append(sans, s)
+			}
+		}
+	case []string:
+		sans = append(sans, oldsans...)
+	}
+
+	if !slices.Contains(sans, addr) {
+		sans = append(sans, addr)
+		log.Infof("%s: added %s to spec.api.sans", h, addr)
+	}
+
+	for i, c := range p.Config.Spec.Hosts.Controllers() {
+		if !slices.Contains(sans, c.Address()) {
+			sans = append(sans, c.Address())
+			log.Infof("%s: added controller %d address %s to spec.api.sans", h, i+1, c.Address())
+		}
+		if c.PrivateAddress != "" && !slices.Contains(sans, c.PrivateAddress) {
+			sans = append(sans, c.PrivateAddress)
+			log.Infof("%s: added controller %d private address %s to spec.api.sans", h, i+1, c.PrivateAddress)
+		}
```

Comment:
I'm not convinced this adds anything of value but it used to be there before v0.19.2

Reply:
Maybe in case of keepalived failover? Or maybe it should add all the virtual addresses?

Reply:
@juanluisvaladas @jnummelin wdyt?

Reply:
> Or maybe it should instead add any virtual addresses?

k0s itself handles that part, no need to k0sctl to worry about.